Stack/Opc.Ua.Core/Types/Encoders/IComplexTypeInstance.cs

```diff
@@ -25,4 +27,29 @@ public interface IComplexTypeInstance
         /// </value>
         ExpandedNodeId TypeId { get; set; }
     }
+
+    /// <summary>
+    /// Interface implemented by complex data type instances that discover and provide type information dynamically
+    /// </summary>
+    public interface IDynamicComplexTypeInstance : IComplexTypeInstance
+    {
+        /// <summary>
+        /// Returns the xml qualified name for the type instance
+        /// </summary>
+        /// <returns>The xml qualified name for the type instance</returns>
+        XmlQualifiedName GetXmlName(IServiceMessageContext context);
+
+        /// <summary>
+        /// Returns the NodeId for the default binary encoding for the type.
+        /// </summary>
+        /// <value>The NodeId for binary encoding.</value>
+        ExpandedNodeId BinaryEncodingId { get; set; }
+
+        /// <summary>
+        /// Returns the NodeId for the default XML encoding for the type.
+        /// </summary>
+        /// <value>The NodeId for the  XML encoding id.</value>
+        ExpandedNodeId XmlEncodingId { get; set; }
```

Comment:
is there no JSON encoding ID needed?

Reply:
Good catch. Actually neither of these are needed - they are already on IEncodeable. I've removed them.